pkg/smokescreen/smokescreen.go

```diff
@@ -148,7 +148,7 @@ func dial(config *Config, network, addr string, userdata interface{}) (net.Conn,
 		outboundHost = v.decision.outboundHost
 	}
 
-	resolved, err := safeResolve(config, network, addr)
+	resolved, _, err := safeResolve(config, network, addr)
```

Comment:
I think we want to update the decision reason here, similar to line 359.  (This code path is hit in the case where we're proxying plain HTTP rather than HTTPS or other traffic via CONNECT.)